nixos/lib/testing-python.nix

```diff
@@ -53,24 +59,23 @@ rec {
         ''
           mkdir -p $out/bin
           cp ${testDriverScript} $out/bin/nixos-test-driver
+          # FIX: exec -a does not work for python shebangs
+          substituteInPlace $out/bin/nixos-test-driver --subst-var name
           chmod u+x $out/bin/nixos-test-driver
           # TODO: copy user script part into this file (append)
 
           wrapProgram $out/bin/nixos-test-driver \
+            --argv0 ${name} \
```

Comment:
It doesn't nothing, but it would be the correct way to do it. I think for this to work throughout the whole nix ecosystem, python would need patching to effectively receive this arg0 from `exec -a`.

Reply:
@tfc this is the rational with regard to your sister comment.